### PR TITLE
Support for arm arm64 and x64

### PR DESCRIPTION
**Pull request by deinok**
Also, this make easy to add other architecture bindings

### CODE REVIEW

File: src/native/Qml.Net.Bindings/Qml.Net.Bindings.props [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    <!-- Extract Buildroot Zip -->
+    <Target Name="BuildrootExtract" DependsOnTargets="BuildrootDownload">
+        <Exec Command="unzip -o ./buildroot-2019.02.zip" 
+            WorkingDirectory="$(MSBuildProjectDirectory)/bin" />
+        <!--<Unzip
+            SourceFiles="$(MSBuildProjectDirectory)/bin/buildroot-2019.02.zip"
+            SkipUnchangedFiles="true"
+            DestinationFolder="$(MSBuildProjectDirectory)/bin/buildroot" />-->
+    </Target>
```
Comment (MaxMommersteeg) [inline src/native/Qml.Net.Bindings/Qml.Net.Bindings.props]: This seems obsolete.

**Comment by pauldotknopf** [comment]
What is the use-case here? Creating native NuGet packages for arm64?

If so, why Buildroot, as opposed to ```apt-get install gcc-aarch64-linux-gnu```?

File: src/native/Qml.Net.Bindings/Qml.Net.Bindings.props (hunk 1 above)
Comment (deinok) [reply src/native/Qml.Net.Bindings/Qml.Net.Bindings.props]: Idealy we should use the Unzip MSBuild task. But works really bad

**Comment by deinok** [comment]
@pauldotknopf Using buildroot have some benefits. Of course the process takes longer than using the package manager.

For example, we get rid of the dependency of having a debian based OS for building. The build is more reproducible. Also we can easy extend this for other triplets that mono support like MIPS and PowerPC.

**Comment by pauldotknopf** [comment]
How does this handle the Qt runtime? Currently, we deploy the pre-built Qt binaries/qml files, delivered by Qt.

**Comment by pauldotknopf** [comment]
I'd like to see the MSBuild ceremony done in C#. See [here]https://github.com/qmlnet/qmlnet/blob/develop/build/scripts/Program.cs).

In my experience, MSBuild isn't *truely* cross platform, and itserror prone. And when it does fail, you are fucked.

**Comment by deinok** [comment]
This just generates the lib required to interop C# to Qt.

So the size of the NuGet is just less than 2MB. This works well for embeded devices where you build the SO with the QT shared libs and then deploy a C# QML app.

Qml.Net.LinuxBinaries could use the generated libraries from buildroot. Also adding support for those arch like for example Qml.Net.LinuxBinaries-linux-arm

**Comment by pauldotknopf** [comment]
Are you deploying this NuGet package (with the ```libQmlNet.so``` compiled from Buildroot) onto an embedded machine/rootfs that was built with Buildroot?

I don't think we can just release a ```libQmlNet.so```, because it won't work with just *any* build of Qt. We depend on ```CONFIG += core-private```, which force us to use the same version of Qt that we compiled against. We plan on removing the need for private headers, see #72.

**Comment by deinok** [comment]
Okey, that explains why it is working for me. Im using the same version in my embedded device.

Seems like the best solution is  #72 menwhile, we could extract the Qt5 from buildroot and put that on .LinuxBinaries package

**Comment by pauldotknopf** [comment]
In your case, I'd recommend just building ```libQmlNet.so``` in your Buildroot and add it to ```/usr/lib```. Then, ```Qml.Net``` will auto discover it and use it.

I don't mind supporting an ARM runtime, but it has to be done using a pattern used at [qmlnet/qt-runtimes](https://github.com/qmlnet/qt-runtimes).

Basically, we'd need to build a ```qt-12.2-{dev|runtime}.tar.gz``` that would be used with for both compiling the ```libQmlNet.so``` and using ```libQmlNet.so``` at runtime.

Let me know if you're interested in doing this. Feel free to close this PR if it makes sense for you.

**Comment by pauldotknopf** [comment]
I'm going to clone this PR now, in favor of creating an ARM runtime over at [qmlnet/qt-runtimes](https://github.com/qmlnet/qt-runtimes).

**Pull request closed without merging**